core/src/main/java/com/mapzen/android/graphics/model/MarkerManager.java

```diff
@@ -27,24 +45,61 @@ public MarkerManager(MapController mapController) {
    */
   public BitmapMarker addMarker(MarkerOptions markerOptions) {
     final Marker marker = mapController.addMarker();
-    marker.setPoint(markerOptions.getPosition());
-    if (markerOptions.getIconDrawable() != null) {
-      marker.setDrawable(markerOptions.getIconDrawable());
-    } else {
-      marker.setDrawable(markerOptions.getIcon());
-    }
-    StyleStringGenerator styleStringGenerator = new StyleStringGenerator();
-    styleStringGenerator.setSize(markerOptions.getWidth(), markerOptions.getHeight());
-    marker.setStylingFromString(styleStringGenerator.getStyleString());
-    return new BitmapMarker(this, marker, styleStringGenerator);
+    BitmapMarker bitmapMarker = bitmapMarkerFactory.createMarker(this, marker,
+        styleStringGenerator);
+    configureMarker(bitmapMarker, markerOptions.getPosition(), markerOptions.getIconDrawable(),
+        markerOptions.getIcon(), markerOptions.getWidth(),
+        markerOptions.getHeight(), markerOptions.isInteractive(), markerOptions.getColorHex(),
+        markerOptions.getColorInt(), markerOptions.isVisible(), markerOptions.getDrawOrder(),
+        markerOptions.getUserData());
+    restorableMarkers.add(bitmapMarker);
```

Comment:
I could be missing something here, but this should be synchronized right? As a when a new scene is loaded its listener should have have a synchronized set of markers to be restored.

Reply:
Good point. Updated to use `Collections.synchronizedList(restorableMarkers)`